src/Management/src/EndpointBase/Health/HealthConverter.cs

```diff
@@ -26,10 +26,17 @@ public override void Write(Utf8JsonWriter writer, HealthEndpointResponse value,
                     writer.WriteString("description", health.Description);
                 }
 
-                foreach (var detail in health.Details)
+                if (health.Details != null && health.Details.Count > 0)
                 {
-                    writer.WritePropertyName(detail.Key);
-                    JsonSerializer.Serialize(writer, detail.Value, options);
+                    writer.WritePropertyName("details");
```

Comment:
Shouldn't this still be detail.Key?

Reply:
No, that merge formatting is misleading.  We still serialize `detail.Key` as we iterate over details object, but now it has an extra nesting with the string "details".